prestashopv1/plugin/modules/indodana/tools/IndodanaTools.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+// phpcs:disable PSR1.Classes.ClassDeclaration.MissingNamespace
+
+require_once(_PS_MODULE_DIR_ . 'indodana' . DIRECTORY_SEPARATOR . 'library' . DIRECTORY_SEPARATOR . 'Indodana' . DIRECTORY_SEPARATOR . 'Payment' . DIRECTORY_SEPARATOR . 'autoload.php');
+
+class IndodanaTools extends Tools
+{
+    public static $indodanaCommon;
+
+    public static function getIndodanaCommon()
+    {
+        if (!isset(self::$indodanaCommon)) {
+            self::$indodanaCommon = new IndodanaCommon\IndodanaCommon([
+                'apiKey' => Configuration::get('INDODANA_API_KEY'),
+                'apiSecret' => Configuration::get('INDODANA_API_SECRET'),
+                'environment' => Configuration::get('INDODANA_ENVIRONMENT'),
+                'seller' => self::getSeller()
+            ]);
+        }
+
+        return self::$indodanaCommon;
+    }
+
+    public static function getSeller()
+    {
+        return [
+            'name' => Configuration::get('INDODANA_STORE_NAME'),
+            'email' => Configuration::get('INDODANA_STORE_EMAIL'),
+            'url' => Configuration::get('INDODANA_STORE_URL'),
+            'address' => [
+                'firstName' => Configuration::get('INDODANA_STORE_NAME'),
+                'lastName' => '',
+                'address' => Configuration::get('INDODANA_STORE_ADDRESS'),
+                'city' => Configuration::get('INDODANA_STORE_CITY'),
+                'postalCode' => Configuration::get('INDODANA_STORE_POSTAL_CODE'),
+                'phone' => Configuration::get('INDODANA_STORE_PHONE'),
+                'countryCode' => Configuration::get('INDODANA_STORE_COUNTRY_CODE')
+            ]
+        ];
+    }
+
+    public static function getTotal($cart)
+    {
+        return $cart->getOrderTotal(true, Cart::BOTH);
+    }
+
+    public static function getShippingFee($cart)
+    {
+        return $cart->getOrderTotal(true, Cart::ONLY_SHIPPING);
+    }
+
+    public static function getDiscount($cart)
+    {
+        $discount = 0;
+        $products = $cart->getProducts();
+        foreach ($products as $product) {
+            if (_PS_VERSION_ >= 1.7) {
+                $discount += $product['reduction'] * $product['quantity'];
+            } else {
+                $discount += ($product['price_without_reduction'] - $product['price_with_reduction'])
+                    * $product['quantity'];
+            }
+        }
+
+        return round($discount);
+    }
+
+    public static function getTax($cart)
+    {
+        $tax = 0;
+        $deficit = 0;
+        $products = $cart->getProducts();
+        foreach ($products as $product) {
+            $tax += ($product['price_without_reduction'] - self::getPriceWithoutReductionWithoutTax($product))
+                * $product['quantity'];
+            $deficit += self::getDeficit($product);
+        }
+
+        return round($tax) + $deficit;
```

Comment:
I'm a bit wary with the roundings, I'm not sure what's the case why we need rounding here

Reply:
Sample case:
tax rate = 10%
product X price = 500000 (with tax)
product X quantity = 1
product X price without tax = 454545.45454545 (value from prestashop return double)
tax only = 45454.545454545 (double)

When we send `checkout` request to Indodana, we will get response:
`Request Exception: {"items":[{"price":{"integer":"Price must be an integer."}},[],{"price":{"integer":"Price must be an integer."}},[]],"sellers":[[]]}`

So I assumed product price, tax and discount must be rounded prior send `checkout` request
CMIIW

Reply:
I've updated the tax calculation, now it doesn't need `getDeficit()` method